pennylane/math/multi_dispatch.py

```diff
@@ -250,6 +252,38 @@ def dot(tensor1, tensor2):
     return np.dot(x, y, like=interface)
 
 
+def get_trainable_indices(values):
+    """Returns a set containing the trainable indices of a sequence of
+    values.
+
+    Args:
+        values (Sequence[tensor_like]): Sequence of tensor-like objects to inspect

```

Comment:
If I [understand correctly](https://docs.python.org/3/library/collections.abc.html), you only need an `Iterable` here, since `enumerate` does not need methods like length or indexing, right?

Reply:
Oh nice catch!
```suggestion
        values (Iterable[tensor_like]): Sequence of tensor-like objects to inspect
```